draft/XXXX-table-metadata/README.md

```diff
@@ -0,0 +1,80 @@
+---
+RFC: XXXX
+Author: Jacob Nesbitt
+Status: draft
+Created: 2020-10-29
+Last-Modified: 2020-10-29
+---
+
+# RFC XXXX: Table/Graph Metadata
+
+Currently, Multinet doesn't store any metadata about tables (or graphs), chiefly type information. This RFC outlines a way to specify, store and retrieve not just type information, but generalized metadata about a table/graph.
+
+## Motivation
+
+The main motivation for this is storing type information for each column in a table. We now have at least two use cases for storing this information.
+
+1. Basic computation/comparison. If we attempt to use these tables for creating graphs and deriving useful data, it's necessary that we store type information, so that we are able to properly compare and compute certain values. For example, if you have a column of numbers, but we store them as strings (the current behavior), then if you try to do comparisons of these column values between different entries, you either will be doing incorrect string comparisons, or will simply not be able to do the comparisons/computations you need to.
+
+2. Providing this information to other applications, so that they do not have to do this data inspection themselves, or if this information is required by those applications. Specifically, [UpSet](https://github.com/VCG/upset) is the first application for which this will be used.
+
+## Proposals
+
+#### Data Model
+There are several ways we could go about this, but in my opinion, the cleanest way is to store this information in a special collection, in the workspace that the table resides in. This special collection would be named something like `multinet_meta`, and each entry would correspond to the metadata on a specific graph or table. I think the data model would look something like this:
```

Comment:
Could there be conflicts if someone uploaded a table called `multinet_meta` since this new collection will reside in the workspace?

Reply:
Unless I'm forgetting something, I don't think this will cause a problem. We don't store tables as collections with the user-specified names; we take their name for the table and stick it in an indirection table that maps to a UUID-looking collection name. So we can still maintain a collection called `multinet_meta` or `metadata` or whatever we like, and still maintain tables with that same name.

Right?

Reply:
We only do this indirection for storing the name of workspaces (as currently they're the only thing we allow to be renamed). For tables/graphs, their names are stored verbatim.

Reply:
Oh shoot, I *was* forgetting something!

I'm fine with just disallowing certain table names by failing during validation (or even just in the form where we supply the table name during upload). We could do name indirection for tables and networks but that would get pretty hairy and for not a lot of benefit.

Reply:
Let's make an issue for this in the client repo as the RFC is settled

Reply:
@AlmightyYakob, so it turns out we can use a [system collection](https://www.arangodb.com/docs/stable/http/collection-creating.html) for this, right?

Reply:
Yes